### PR TITLE
Clarify detached JWS rules per comments in #556

### PR DESCRIPTION
**Pull request by dhh1128**
See [this comment](https://github.com/hyperledger/aries-rfcs/issues/556#issuecomment-710676317) on issue #556.

Signed-off-by: Daniel Hardman <daniel.hardman@gmail.com>

### CODE REVIEW

**Review by swcurran — APPROVED** [review]

File: concepts/0017-attachments/README.md [hunk 1]
```diff
@@ -346,10 +346,7 @@ signing key when the sender is performing key rotation.
 
 Embedded and appended attachments support signatures by the addition of a `data.jws` field
 containing a signature in [JWS (RFC 7515) format](https://tools.ietf.org/html/rfc7515)
-with [Detached Content](https://tools.ietf.org/html/rfc7515#appendix-F).
-The payload of the JWS is the raw data of the attachment, whether externally referenced
-or encoded in base64 format, and is not contained within the signature itself.
-Signatures over inlined JSON attachments are not currently defined as this
+with [Detached Content](https://tools.ietf.org/html/rfc7515#appendix-F). The payload of the JWS is the raw bytes of the attachment, appropriately base64url-encoded per JWS rules. If these raw bytes are incorporated by value in the DIDComm message, they are already base64url-encoded in `data.base64` and are thus directly substitutable for the suppressed `data.jws.payload` field; if they are externally referenced, then the bytes must be fetched via the URI in `data.links` and base64url-encoded before the JWS can be fully reconstituted. Signatures over inlined JSON attachments are not currently defined as this
```
Comment (kdenhartog) [inline concepts/0017-attachments/README.md:349]: slightly pedantic (but would impact our interop) do we state that the base64 property is actually base64url anywhere?

**Review by kdenhartog — APPROVED** [review]
Looks good and noticed the potential for different interpretations around the base64 parameter. We should make sure this doesn't cause us any headaches by making sure this is explicit consistently (thinking it may be left unstated in 0017, but that's also a concept not necessarily a feature)

**Review by llorllale — APPROVED** [review]

File: concepts/0017-attachments/README.md (hunk 1 above)
Comment (llorllale) [reply concepts/0017-attachments/README.md:349]: Attachments RFC still defines `base64` to be just base64-encoding: https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0017-attachments/README.md#attachment-descriptor-structure

Comment (llorllale) [reply concepts/0017-attachments/README.md:349]: +1 to using base64url encoding everywhere, and updating the meaning of the attachment's `base64` property.

Comment (TelegramSam) [reply concepts/0017-attachments/README.md:349]: Discussed in WG 20201104, We should be consistent with our use of Base64URL vs Base64.

Comment (dhh1128) [reply concepts/0017-attachments/README.md:349]: I don't understand. I agree with the WG decision, and with @llorllale 's +1 comment. But the PR I raised is already explicit about base64url encoding. So is someone asking that this PR be fixed in some way, or that a new PR be raised that applies additional fixes around base64url encoding in places that this PR didn't touch?

Comment (swcurran) [reply concepts/0017-attachments/README.md:349]: The request is for a check to be done that base64/base64url references in the RFC are appropriate.  Could impact this PR or trigger another.

Comment (dhh1128) [reply concepts/0017-attachments/README.md:349]: Okay, I have updated RFC 0017 to be totally clear and consistent about base64url throughout.

Work to update other RFCs to be consistent is embodied in a different PR, #561. That PR is more controversial. Let's get this one merged and then discuss the other one.

**Comment by dhh1128** [comment]
@TelegramSam or @swcurran : Can we get this merged, just for hygiene's sake?

**Comment by swcurran** [comment]
@dhh1128 -- can you please resolve the conflicts?

**Pull request merged**